plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordCursor.java

```diff
@@ -135,8 +141,29 @@ public boolean advanceNextPosition()
 
         try {
             if (resultSet == null) {
-                log.debug("Executing: %s", statement.toString());
-                resultSet = statement.executeQuery();
+                try {
+                    Future<ResultSet> resultSetFuture = executor.submit(() -> {
+                        log.debug("Executing: %s", statement.toString());
+                        return statement.executeQuery();
+                    });
+                    // executeQuery() may block uninterruptedly, using async way so we are able to cancel remote query
```

Comment:
```suggestion
                    // statement.executeQuery() may block uninterruptedly, using async way so we are able to cancel remote query
```

Reply:
also, worth adding some URL / pointer

Reply:
To something that would explain why statement.executeQuery() may block uninterruptedly